### DISCUSSION TITLE
interupt feature for gpio_mcp23xxx

### DISCUSSION THREAD

**Discussion by JosephBerman** [zephyrproject-rtos/zephyr#51914] [Q&A]
**Is your enhancement proposal related to a problem? Please describe.**
Battery drain is higher because it is polling for keyboard button presses as opposed to an interrupt. Correct me if I'm wrong but theres not interrupt functions set up currently for it right now. In the datasheet it has some interrupt settings. 

**Describe the solution you'd like**
I'm going to make the code and test it out. Will submit a PR once it is completed.  Let me know if this is reasonable/already has been done. I'm not that familiar with the device, coding this for a friend.

**Comment by arbrauns** [comment]
@JosephBerman did you get anywhere with the implementation? I need this feature as well, and it would be great to start even from a half-finished attempt.
